tests/main/ubuntu-core-refresh/task.yaml

```diff
@@ -0,0 +1,81 @@
+summary: Check that the ubuntu-core system is rebooted after the core snap is refreshed
+
+details: |
+    This test checks that when invoking a manual refresh/revert for core or core18 snaps,
+    a reboot is triggered and the command would exit after the first phase of the installation
+    reporting "snapd is about to reboot the system"
+
+systems: [ubuntu-core-1*]
+
+execute: |
+    #shellcheck source=tests/lib/journalctl.sh
+    . "$TESTSLIB"/journalctl.sh
+    #shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB"/systems.sh
+
+    TARGET_SNAP_NAME=core
+    if is_core18_system; then
+        TARGET_SNAP_NAME=core18
+    fi    
+
+    # After installing a new version of the core/core18 snap the system is rebooted
+    if [ "$SPREAD_REBOOT" = 0 ]; then
+        snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}" > initialRev    
+
+        # use journalctl wrapper to grep only the logs collected while the test is running
+        if get_journalctl_log | MATCH "Waiting for system reboot"; then
+            echo "Already waiting for system reboot, exiting..."
+            exit 1
+        fi
+
+        # install new target snap
+        snap download "$TARGET_SNAP_NAME"
```

Comment:
Can we just install the same revision we have? What's the point of installing a snap we downloaded from the network here?

Reply:
It won't be cached if core/core18 is published at around the moment this runs. I think some tests already install the locally installed version of a snap this way so we could try to reuse that code.

I'm also worried about snapd figuring out that a snap provided as a file actually matches a known assertion that would give it a real revision.